define/types.go

```diff
@@ -178,11 +178,22 @@ func TempDirForURL(dir, prefix, url string) (name string, subdir string, err err
 	return "", "", errors.New("unreachable code reached")
 }
 
-func cloneToDirectory(url, dir string) ([]byte, string, error) {
+// ParseGitBuildContext parses git build context to `repo`, `sub-dir`
+// `commit` and `branch`, accepts GitBuildContextin the format of
+// `repourl.git#branch:subdir?commit`.
+//
+// TODO: once following package `define` is refactored this function
```

Comment:
As noted here pkg `define` needs a refactor so that functions like `ParseGitBuildContext` can be moved to relevent package. But since its a code move around of lot of existing function in `define/types.go` so will do that in a followup PR after this PR in order to make sure review is easy.

Reply:
Why export `ParseGitBuildContext()` at all, if it's only going to be called from one place that's in the same package?

Reply:
Made it private.